src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +586,32 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static bool ValidateArguments(string[] args, CommandLineApplication app)
+    {
+        // Get all valid options from the configuration files
+        var validArguments = app.GetOptions().SelectMany(o => new[] { o.LongName, o.ShortName }).ToArray();
+
+        var argumentsNamesProvided = GetArgumentNames(args).Select(a => a.ToString()).ToArray();
```

Comment:
You don't need to allocate strings, `MemoryContentsComparer` can compare `Memory<char>` directly without additional allocations

Reply:
Avoid `ToArray`, either keep it `IEnumerable` or use `ToPooledList` and use pooled memory (need to be disposed with `using` statement).